docs/changelog.rst

```diff
@@ -37,6 +37,24 @@ See :ref:`upgrade-guide-2.6` for help upgrading to the new version.
 * ``py::memoryview``  update and documentation.
   `#2223 <https://github.com/pybind/pybind11/pull/2223>`_
 
+* The Python package was reworked to be more powerful and useful.
+  `#2433 <https://github.com/pybind/pybind11/pull/2433>`_
+
+  * :ref:`build-setuptools` is easier thanks to a new
+    ``pybind11.setup_helpers`` module, which provides utilities to use
+    setuptools with pybind11. It can be used via PEP 518, ``setup_requires``,
+    or by directly copying ``setup_helpers.py`` into your project.
+
+  * CMake configuration files are now included in the Python package. Use
+    ``pybind11.get_cmake_dir()`` or ``python -m pybind11 --cmakedir`` to get
+    the directory with the CMake configuration files, or include the
+    site-packages location in your ``CMAKE_MODULE_PATH``. Or you can use the
+    new ``pybind11[global]`` option, which installs the CMake files and headers
```

Comment:
How does one use this "option"? I am still really confused about what this even means, and what the difference between ``pybind11`` and ``pybind11global`` is.

Reply:
Wherever you install or list `pybind11` as a package, you can list `pybind11[global]` instead (though you need quotes on the command line if you are in cmd.exe). So in `setup.py`, `requirements.txt`, `Pipfile`, `pyproject.toml`, etc.

`pybind11` only places files in `<ENVROOT>/lib/pythonX.X/site-packages/pybind11`. `include` and `share` are added to the pybind11 folder. (This was done in pybind11 2.4 or 2.5, and solved a lot of search issues)

`pybind11-global` only places includes and cmake files in  (no python files) in `<ENVROOT>/include` and `<ENVROOT>/share`. If `<ENVROOT>` is your system root, this is a little ugly. However, for pyproject.toml, this is always a venv, and it's fine in venvs, conda, and anywhere that is not your system root. (This is manually hacked in by the conda-forge recipe currently)

> Mostly for backward compatibility, `pybind11-global` also puts a copy of the pybind11 headers in the official "headers" slot, something like `include/pythonX.X<m>/`, but this is very hard to even find (it's the reason behind the old "user=True/False` setting, since it's not the same for a user install vs. a normal install). This was the only place they were put pre 2.4 or 2.5.